apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -587,11 +597,15 @@ async function traverseQuestionAndExecuteFunctions(phase, codeCaller, data, cont
       if (phase === 'render' && !renderedElementNames.includes(elementName)) {
         renderedElementNames.push(elementName);
       }
-      // Populate the extensions used by this element.
-      data.extensions = [];
-      if (Object.prototype.hasOwnProperty.call(context.course_element_extensions, elementName)) {
-        data.extensions = context.course_element_extensions[elementName];
-      }
+
+      // Populate any extensions used by this element.
+      const extensions = run(() => {
+        if (Object.prototype.hasOwnProperty.call(context.course_element_extensions, elementName)) {
+          return context.course_element_extensions[elementName];
+        }
+        return {};
```

Comment:
Slight change: previously, the default was an empty array, but now it's an empty object. This is consistent with the type that's used when extensions are present, as well as the Python implementation of all this:

https://github.com/PrairieLearn/PrairieLearn/blob/2743de3be2e5e241522bceaf5f614df5f0711539/apps/prairielearn/python/prairielearn/internal/question_phases.py#L222